hal/spi_api.h

```diff
@@ -33,6 +33,7 @@
 #define SPI_EVENT_INTERNAL_TRANSFER_COMPLETE (1 << 30) // Internal flag to report that an event occurred
 
 #define SPI_FILL_WORD         (0xFFFF)
+#define SPI_FILL_CHAR         (0xFF)
```

Comment:
'SPI::write' buffers are of char* type, and SPI SD driver support only 8-bit SPI (at present).
I was getting build warnings for both above cases. Instead of adding define only for SD driver added generic one.